client/src/components/CheckoutForm/CheckoutForm.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-unused-vars */
+import React, { useState } from 'react';
+import { Formik, Form } from 'formik';
+import { Stepper, Step, StepLabel, Button } from '@mui/material';
+import { UserDetailsForm, ShippingAddressForm, PaymentForm, PaymentSuccess, CheckoutSummary } from '..';
+import checkoutFormModel from '../../utils/checkoutFormModels/checkoutFormModel';
+import CheckoutFormInitialValues from '../../utils/checkoutFormModels/CheckoutFormInitialValues';
+import CheckoutFormvalidationSchema from '../../utils/checkoutFormModels/CheckoutFormvalidationSchema';
+
+const steps = ['User Details', 'Shipping Address', 'Payment Details'];
+
+const CheckoutForm = () => {
+  const [activeStep, setActiveStep] = useState(0);
+  const lastStep = steps.length - 1;
+  const currentValidationSchema = CheckoutFormvalidationSchema[activeStep];
+
+  const GoToNextStep = () => {
+    setActiveStep(activeStep + 1);
+  };
+
+  const GoTOPrevStep = () => {
+    setActiveStep(activeStep - 1);
```

Comment:
suggest using previous value of active step:
```suggestion
    setActiveStep(prev => prev - 1);
```

Reply:
call back used